irohad/validation/utils.hpp

```diff
@@ -51,12 +51,12 @@ namespace iroha {
     inline bool peersSubset(
         const shared_model::interface::types::SignatureRangeType &signatures,
         const Peers &peers) {
-      return signaturesSubset(signatures,
-                              peers
-                                  | boost::adaptors::transformed(
-                                        [](const auto &p) -> decltype(auto) {
-                                          return p->pubkey();
-                                        }));
+      return signaturesSubset(
+          signatures,
+          peers | boost::adaptors::transformed([](auto const &p) {
+            return shared_model::interface::types::PublicKeyHexStringView{
+                p->pubkey()};
```

Comment:
```suggestion
            using shared_model::interface::types::PublicKeyHexStringView;
            return PublicKeyHexStringView{p->pubkey()};
```
just for readability